### ISSUE TITLE
TLS: Store TLS cert-key secrets in a dedicated namespaces

### ISSUE THREAD

**Issue by hbagdi** [kubernetes-sigs/service-apis#105]
**What would you like to be added**:

As a Kubernetes operator
- I want to a specific namespace to contain all TLS secrets and control access to that namespace
- I don't want Kubernetes Ingress Controllers to have read access to all secrets in my cluster. I would prefer to give read access to all secrets in a specific namespace.

**Why is this needed**:

Lock down permissions for humans and controllers.

The solution is proposed here part of the user-story itself and maybe the story could be written better. k8s RBAC model doesn't allow one to become more fine-grained than this.

**Comment by bowei** [comment]
It seems like this can be implement by Gateway namespace restrictions.

Can we have a cookbook example to illustrate this?

**Comment by hbagdi** [comment]
> It seems like this can be implement by Gateway namespace restrictions.

That's correct.

It seems we are going towards the world where TLS information will be present in the Gateway resource. In the current model, one can control which namespaces can have gateway resource via GatewayClass. This user-story is covered as an extension of this where the namespaces containing the Gateway will be the only ones containing the TLS secrets. 

I will add a cookbook example once we have TLS definitions merged into master.
@bowei Can you assign this to me?

**Comment by jpeach** [comment]
`CertificateObjectReference` is a local reference, so you can't use secrets that are not in the same namespace as the Gateway. I think that this should be allowed as long as this is a reference and we don't give TLS keys inline.

**Comment by hbagdi** [comment]
I'm waiting on consensus around #193. Once we have that, I can propose this as a PR.

**Comment by hbagdi** [comment]
Since Gateway can live in one namespace and TLS can be defined at the Gateway layer, I think this can be closed.

@Miciah @robscott @bowei @jpeach Thoughts?

**Comment by hbagdi** [comment]
Please re-open if needed.

/close
